From 93730160f9e9987d31cc5e1757fa774bcc964ea8 Mon Sep 17 00:00:00 2001
From: Dario Binacchi <dario.binacchi@amarulasolutions.com>
Date: Sun, 10 May 2026 10:36:55 +0200
Subject: [PATCH] ARM: dts: stm32: add FWU metadata node for
 stm32mp157f-dk2

The FWU metadata structure is accessed through the driver model
interface. While commit a402adc664d4 ("stm32mp1: Add a node for the
FWU metadata device") added this support for the stm32mp157c variants,
it was missing for the stm32mp157f-dk2 board.

Since the FWU metadata is also stored on the uSD card for the
stm32mp157f-dk2 board, add the fwu-mdata node in its u-boot specific
dtsi file to allow accessing the metadata structure properly.

Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
Upstream-Status: Pending [specific to ST boards]
---
 arch/arm/dts/stm32mp157f-dk2-u-boot.dtsi | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/dts/stm32mp157f-dk2-u-boot.dtsi b/arch/arm/dts/stm32mp157f-dk2-u-boot.dtsi
index 53133e98ba82..329c48723a97 100644
--- a/arch/arm/dts/stm32mp157f-dk2-u-boot.dtsi
+++ b/arch/arm/dts/stm32mp157f-dk2-u-boot.dtsi
@@ -5,6 +5,13 @@
 
 #include "stm32mp157a-dk1-u-boot.dtsi"
 
+/ {
+	fwu-mdata {
+		compatible = "u-boot,fwu-mdata-gpt";
+		fwu-mdata-store = <&sdmmc1>;
+	};
+};
+
 &sdmmc2 {
 	status = "disabled";
 };
-- 
2.43.0

